libs/beacon/src/beacon_service.cpp

```diff
@@ -487,37 +500,43 @@ BeaconService::State BeaconService::OnVerifySignaturesState()
   }
 
   // Don't lock until the promise has resolved! Otherwise the system can deadlock.
-  std::lock_guard<std::mutex> lock(mutex_);
-
-  if (ret.threshold_signatures.empty())
   {
-    FETCH_LOG_INFO(LOGGING_NAME, "Peer wasn't ready when asking for signatures: ",
-                   qual_promise_identity_.identifier().ToBase64());
-    state_machine_->Delay(std::chrono::milliseconds(10));
-    return State::COLLECT_SIGNATURES;
-  }
+    std::lock_guard<std::mutex> lock(mutex_);
```

Comment:
Only change here is to not lock when doing manager verify since the lock stops peers from pulling the sigs

Reply:
can't tell if sarcastic but sure